File: 프로그래머스/0/120894. 영어가 싫어요/영어가 싫어요.cpp
#include <string>
#include <vector>
using namespace std;

long long solution(string numbers) {
    vector<string> nums = {"zero", "one", "two", "three", "four", "five", 
                          "six", "seven", "eight", "nine"};
    string answer = "";
    string temp = "";
    
    for(char c : numbers) {
        temp += c;
        for(int i = 0; i < nums.size(); i++) {
            if(temp == nums[i]) {
                answer += to_string(i);
                temp = "";
                break;
            }
        }
    }
    
    return stoll(answer);
}

File: 프로그래머스/0/120894. 영어가 싫어요/영어가 싫어요.swift
import Foundation

func solution(_ numbers:String) -> Int64 {
    let num: [String] = [
        "zero", "one", "two", "three",
        "four", "five", "six", "seven", "eight", "nine"
    ]
    var str: String = ""
    var result: String = ""
    for i in numbers {
        str += String(i)
        for (offset, element) in num.enumerated() {
            if str == element {
                result.append(String(offset))
                str = ""
                break
            }
        }
    }
    return Int64(result)!
}